staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -161,22 +162,37 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 	return unmarshalToObject(s.typer, s.creater, &actual, into, unk.Raw)
 }
 
+// EncodeWithAllocator writes an object to the provided writer.
+// In addition, it allows for providing a memory allocator for efficient memory usage during object serialization.
+func (s *Serializer) EncodeWithAllocator(obj runtime.Object, w io.Writer, memAlloc runtime.MemoryAllocator) error {
+	return s.encode(obj, w, memAlloc)
+}
+
 // Encode serializes the provided object to the given writer.
 func (s *Serializer) Encode(obj runtime.Object, w io.Writer) error {
+	return s.encode(obj, w, nil)
+}
+
+func (s *Serializer) encode(obj runtime.Object, w io.Writer, memAlloc runtime.MemoryAllocator) error {
 	if co, ok := obj.(runtime.CacheableObject); ok {
-		return co.CacheEncode(s.Identifier(), s.doEncode, w)
+		return co.CacheEncode(s.Identifier(), func(obj runtime.Object, w io.Writer) error { return s.doEncode(obj, w, memAlloc) }, w)
 	}
-	return s.doEncode(obj, w)
+	return s.doEncode(obj, w, memAlloc)
 }
 
-func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
+func (s *Serializer) doEncode(obj runtime.Object, w io.Writer, memAlloc runtime.MemoryAllocator) error {
 	prefixSize := uint64(len(s.prefix))
 
 	var unk runtime.Unknown
 	switch t := obj.(type) {
 	case *runtime.Unknown:
 		estimatedSize := prefixSize + uint64(t.Size())
-		data := make([]byte, estimatedSize)
+		var data []byte
+		if memAlloc != nil {
```

Comment:
Instead of this, can we add sth like `SimpleAllocator` which effectively will be always allocating the needed memory.

And just pass instead when non memAlloc is passed?

Reply:
we could do that, would it be okay to create `SimpleAllocator` inside this `protobuf` package?

Reply:
an alternative would be to create a new package `allocator` and move the `SimpleAllocator` and the `Allocator` there. Doing so would allow it to be reused.

Reply:
I would personally put it in allocator.go that you're already adding.
It seems as simple as:

```
type SimpleAlocator struct{}{}

func (*SimpleAlocator) Allocate(n uint64) []byte {
  return make([]byte, n, n)
}
```

Reply:
putting it into `allocator.go` would create an import cycle

Reply:
creating a new package would prevent it

Reply:
I don't understand - import cycle with what?

Allocator is not importing anything. If we don't have import cycle problem now, adding a second implementation of the same interface in the same file won't cause any new problems.